PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+            <#
+                One can use curly braces to give a range of numbers. This one means "between 2 and 4
+                of any character"
+            #>
+            'Paper airplane' -match '.{2,4}'
+    
+            $fruitMatch = 'p____'
+            [regex]::Matches('Apples', $fruitMatch).Value | Should -Be 'pp'
+            $fruitMatch.Contains('}') | Should -BeTrue     # Make sure you use curly braces!
+    
+            <#
+                If you omit the second number, the custom quantifier becomes "this many or more".
+                This one means "two or more non-whitespace characters"
+            #>
+            [bool]$whitespaceMatch = $____ # Should be either $true or $false
+            $whitespaceMatch | Should -Be 'The moon in June is a big balloon' -match '\s{2,}'
+        }
+    }
+
+    Context '( and ) - Round Braces' {
+        It 'groups patterns together' {
+            <#
+                Round brackets are used to create groups of symbols. These groups can then have
+                quantifiers applied to the entire group, or be used in approximately 42 trillion
+                other ways. This is just the tip of the round bracket/regular expression groups
+                iceberg.
+            #>
+    
+            '____' | Should -Be [regex]::Matches('Bears Beat Bongos', '(B.+){3}').Value
+    
+            $badgers = 'BadgerBadgerBadger'
+            $groupingPattern = '____'
+            [regex]::Matches($badgers, $groupingPattern).Value | Should -Be 'BadgerBadgerBadger'
+            $groupingPattern.Contains(')') | Should -BeTrue    # Use a group!
+    
+        }
+    }
+
+    Context '[ and ] - Square Brackets' {
+        It 'defines a set' {
+            <#
+                Square brackets denote a set, or array/collection, of symbols within a regular
+                expression. Imagine a pattern that might read "a or b or c or d". You may use square
+                brackets to create that regex. The set contained within the square brackets
+                represent one charactermwithin the string being searched.
+            #>
+    
+            [bool]$eolMatch = $____ # Should be either $true or $false
+            $eolMatch | Should -Be 'End of the line' -match '[efg]$'
+    
+            '____' -match '[abcdefg]$' | Should -BeTrue
+    
+            <#
+                Normally, the ^ character means "start of a line", but inside of square brackets, it
+                negates the set. Instead of "these symbols" it means "not these symbols"
+            #>
+            '____' -match '[^abcdefg]$' | Should -BeTrue
+    
+            <#
+                What do you think you might do if you want to match a literal ^ symbol within a set?
+                Use the ^ (caret) symbol in this one.
+            #>
+            $caretMatch = '[____]'
+            '^&*' -match $caretMatch | Should -BeTrue
+            $caretMatch.Contains('^') | Should -BeTrue      # Use the caret symbol!
+        }
```

Comment:
Same here, spacing, separate `It` blocks.

Also, probably worth using the proper name for this ("character class")